doc/Gemm.md

```diff
@@ -0,0 +1,405 @@
+
+The notable calls in the stack are (all located in `src/interface/gemm_interface.hpp`):
+
+- `blas::internal::_gemm`
+
+  - calls `_gemm_backend()` always passing `strided` for the `gemm` batch type (as the interleaved kernel is intended only for batch operations). The `_batch_gemm` call instead passes the batch type through.
+
+- `blas::internal::_gemm_backend()`
+
+  - calls `_gemm_is_beta_zero` with different transpose template parameters depending on the runtime values passed.
+
+- `blas::internal::_gemm_is_beta_zero()`
+
+  - calls `_gemm_platform_specific` depending on whether beta == 0 or not.
+
+- `blas::internal::_gemm_platform_specific`
+
+  - calls `blas::gemm::backend::_gemm` which is the backend target specific GEMM.
+
+## GEMM Backends
+
+GEMM backends are a mechanism to provide different compile-time configurations for different hardware platforms/backends. Backend selection is controlled by passing the cmake variable `TARGET` during CMake configuration, for example passing `-DTARGET=INTEL_GPU` would select the appropriate configurations for Intel GPUs. This cmake variable causes a corresponding define for the selected platform to be included in the source which then controls backend selection through `#ifdef`s in `src/interface/blas3/backend/backend.hpp` like so:
+
+```c++
+#if defined(RCAR)
+#include "interface/blas3/backend/rcar.hpp"
+#elif defined INTEL_GPU
+#include "interface/blas3/backend/intel_gpu.hpp"
+#elif defined AMD_GPU
+#include "interface/blas3/backend/amd_gpu.hpp"
+#elif defined ARM_GPU
+#include "interface/blas3/backend/arm_gpu.hpp"
+#elif defined POWER_VR
+#include "interface/blas3/backend/power_vr.hpp"
+#else
+#include "interface/blas3/backend/default_cpu.hpp"
+#endif
+```
+
+These backend headers call `Gemm_Launcher::_select_gemm()` with various parameters depending on the inputs given. For example, they commonly call different configurations depending on input size to obtain optimal performance for a given size or range of sizes. Backend configurations are covered in further detail in [this section](#backend-configurations).
+
+## GEMM Launcher
+
+The `Gemm_Launcher` class wraps the creation of the actual `Gemm` class as well as the creation of the matrix views (which are what is actually passed to the `Gemm` class for use in the kernel). This happens in the `::select_gemm()` member function where it also executes the created `Gemm` through the passed in executor and returns the associated event.
+
+```c++
+namespace blas {
+
+/*!
+ * @brief Wrapper around Gemm. Creates the views, then makes and launches Gemm
+ */
+template <int WgSize, bool DoubleBuffer, bool ConflictA, bool ConflictB, 
+          int ClSize, typename TileT, bool TransA, bool TransB,
+          int GemmMemoryType, int GemmAlgorithm, int GemmVectorization,
+          bool is_beta_zero, int VectorSize, int BatchType>
+
+template <typename Executor, typename container_t0, typename container_t1, 
```

Comment:
Is this second `template` required?

Reply:
I believe so yes, at least according to https://en.cppreference.com/w/cpp/language/member_template